src/databricks/labs/ucx/assessment/aws.py

```diff
@@ -0,0 +1,251 @@
+import csv
+import dataclasses
+import io
+import json
+import logging
+import re
+import subprocess
+import typing
+from collections.abc import Callable, Iterable
+from dataclasses import dataclass
+from functools import lru_cache, partial
+
+from databricks.labs.blueprint.parallel import Threads
+from databricks.sdk import WorkspaceClient
+from databricks.sdk.service.workspace import ImportFormat
+
+logger = logging.getLogger(__name__)
+
+
+@dataclass
+class AWSRole:
+    path: str
+    role_name: str
+    role_id: str
+    arn: str
+
+
+@dataclass
+class AWSPolicyAction:
+    resource_type: str
+    action: set[str]
+    resource_path: str
+
+
+@dataclass
+class AWSInstanceProfileAction:
+    instance_profile_arn: str
+    resource_type: str
+    actions: str
+    resource_path: str
+    iam_role_arn: str | None = None
+
+
+@dataclass
+class AWSInstanceProfile:
+    instance_profile_arn: str
+    iam_role_arn: str | None = None
+
+    ROLE_NAME_REGEX = r"arn:aws:iam::[0-9]+:(?:instance-profile|role)\/([a-zA-Z0-9+=,.@_-]*)$"
+
+    @property
+    def role_name(self):
+        if self.iam_role_arn:
+            arn = self.iam_role_arn
+        else:
+            arn = self.instance_profile_arn
+        role_match = re.match(self.ROLE_NAME_REGEX, arn)
+        if not role_match:
+            logger.error(f"Role ARN is mismatched {self.iam_role_arn}")
+            return None
+        else:
+            return role_match.group(1)
+
+
+@lru_cache(maxsize=1024)
+def run_command(command):
+    logger.info(f"Invoking Command {command}")
+    process = subprocess.Popen(command, stdout=subprocess.PIPE, stderr=subprocess.PIPE, shell=True)  # noqa: S602
+    output, error = process.communicate()
+    return process.returncode, output.decode("utf-8"), error.decode("utf-8")
+
+
+class AWSResources:
+    S3_ACTIONS: typing.ClassVar[set[str]] = {"s3:PutObject", "s3:GetObject", "s3:DeleteObject", "s3:PutObjectAcl"}
```

Comment:
```suggestion
    S3_ACTIONS: typing.ClassVar[set[str]] = {"s3:PutObject", "s3:GetObject", "s3:DeleteObject"}
```
I think put obj acl is irrelevant

Reply:
It is in our docs. I'm not sure either.